compiler/rustc_infer/src/infer/mod.rs

```diff
@@ -1598,22 +1605,29 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
     ///
     /// This handles inferences variables within both `param_env` and `substs` by
     /// performing the operation on their respective canonical forms.
+    #[instrument(skip(self), level = "debug")]
     pub fn const_eval_resolve(
         &self,
         param_env: ty::ParamEnv<'tcx>,
         unevaluated: ty::Unevaluated<'tcx>,
         span: Option<Span>,
     ) -> EvalToConstValueResult<'tcx> {
         let substs = self.resolve_vars_if_possible(unevaluated.substs);
+        debug!(?substs);
 
         // Postpone the evaluation of constants whose substs depend on inference
         // variables
         if substs.has_infer_types_or_consts() {
-            return Err(ErrorHandled::TooGeneric);
+            debug!("substs have infer types or consts: {:?}", substs);
+            if substs.has_infer_types_or_consts() {
+                return Err(ErrorHandled::TooGeneric);
+            }
```

Comment:
```suggestion
            return Err(ErrorHandled::TooGeneric);
```
what's happening here?